operators/src/processing/raster_kernel/kernel_function.rs

```diff
@@ -0,0 +1,208 @@
+use super::{error, RasterKernelError};
+use crate::util::{number_statistics::NumberStatistics, Result};
+use geoengine_datatypes::raster::{Grid2D, GridShape2D, GridSize, Pixel};
+use num::Integer;
+use num_traits::AsPrimitive;
+use snafu::ensure;
+
+/// A function that calculates one pixel values from a neighborhood of pixels.
+pub trait KernelFunction<P>: Sync + Send + Clone
```

Comment:
wouldn't it make more sense to call it MatrixFunction?

> A kernel is a matrix, which is slid across the image and multiplied with the input such that the output is enhanced in a certain desirable manner.

but that isn't the case for the standarddeviation, which ich a function on a Matrix, but is not a kernel

Reply:
changed the name to neighborhoodaggregate